src/Illuminate/Routing/ResponseFactory.php

```diff
@@ -161,6 +162,20 @@ public function streamDownload($callback, $name = null, array $headers = [], $di
         return $response;
     }
 
+    /**
+     * Directly output stream as an HTTP Response.
+     *
+     * @param  StreamInterface  $stream
+     * @param  string  $name
+     * @return StreamedResponse
+     */
+    public function directStreamDownload(StreamInterface $stream, string $name)
+    {
+        return $this->streamDownload(function () use ($stream) {
+            echo $stream;
```

Comment:
If the file is too big, wouldn't this load it all into memory? As pointed on the PSR docs?

https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/StreamInterface.php#L20

Isn't the idea of outputting a stream the possibility of chunking its output? Something like this:

~~~php
return $this->streamDownload(function () use ($stream) {
    if ($stream->isSeekable()) {
        $stream->rewind();
    }

    while (! $stream->eof()) {
        echo $stream->read(4096);
    }
});
~~~

Reply:
Or with `fpassthru($stream->detach())` instead of the `while` loop.